ggml/src/ggml-cuda/ggml-cuda.cu

```diff
@@ -1200,7 +1200,7 @@ static void ggml_cuda_op_mul_mat_cublas(
 
     const bool use_fp16 = (src0->type == GGML_TYPE_F16 || ggml_is_quantized(src0->type)) && ggml_is_contiguous(src0) && row_diff == src0->ne[1] && dst->op_params[0] == GGML_PREC_DEFAULT;
 
-    if (src0->type == GGML_TYPE_BF16 && ggml_is_contiguous(src0) && row_diff == src0->ne[1]) {
+    if ((GGML_CUDA_CC_IS_NVIDIA(cc) || GGML_CUDA_CC_IS_AMD(cc) || (GGML_CUDA_CC_IS_MTHREADS(cc) && cc >= GGML_CUDA_CC_QY2)) && src0->type == GGML_TYPE_BF16 && ggml_is_contiguous(src0) && row_diff == src0->ne[1]) {
```

Comment:
I would suggest adding something like `const bool XXX = ...` on the line above. As it is it's not clear to me what the correct interpretation of the check is. (Also it prevents the line from becoming too long.)

Reply:
Thanks for the review! I've added `support_bf16` and `support_fp16` consts. Previously, line 1203 didn't perform any compute capability check, so MTT S80 would go into that code path.